Lib/asyncio/streams.py

```diff
@@ -766,18 +391,17 @@ def get_extra_info(self, name, default=None):
             if exc is not None:
                 raise exc
         if self._transport.is_closing():
-            # Yield to the event loop so connection_lost() may be
-            # called.  Without this, _drain_helper() would return
-            # immediately, and code that calls
-            #     write(...); await drain()
-            # in a loop would never call connection_lost(), so it
-            # would not see an error when the socket is closed.
-            await tasks.sleep(0, loop=self._loop)
+            # Wait for protocol.connection_lost() call
+            # Raise connection closing error if any,
+            # ConnectionResetError otherwise
+            await sleep(0)
```

Comment:
Change made by @asvetlov that was not related to the streaming API. It fixes an issue that prevented the call to `protocol.connection_lost()`. See comment for details.
```suggestion
            # Yield to the event loop so connection_lost() may be
            # called.  Without this, _drain_helper() would return
            # immediately, and code that calls
            #     write(...); await drain()
            # in a loop would never call connection_lost(), so it
            # would not see an error when the socket is closed.
            await tasks.sleep(0, loop=self._loop)
```
Should we restore this? I don't see an issue with doing so or a reason why it should be removed.

Reply:
Yes, I'll restore it. Thanks!